client/homebrew/pages/vaultPage/vaultPage.jsx

```diff
@@ -0,0 +1,478 @@
+require('./vaultPage.less');
+
+const React = require('react');
+const { useState, useEffect, useRef } = React;
+const cx = require('classnames');
+
+const Nav = require('naturalcrit/nav/nav.jsx');
+const Navbar = require('../../navbar/navbar.jsx');
+const RecentNavItem = require('../../navbar/recent.navitem.jsx').both;
+const Account = require('../../navbar/account.navitem.jsx');
+const NewBrew = require('../../navbar/newbrew.navitem.jsx');
+const HelpNavItem = require('../../navbar/help.navitem.jsx');
+const BrewItem = require('../basePages/listPage/brewItem/brewItem.jsx');
+const SplitPane = require('../../../../shared/naturalcrit/splitPane/splitPane.jsx');
+
+const request = require('../../utils/request-middleware.js');
+
+const VaultPage = (props) => {
+	const [title, setTitle] = useState(props.query.title || '');
+	//state author
+	const [author, setAuthor] = useState(props.query.author || '');
+	const [legacy, setLegacy] = useState(props.query.legacy !== 'false');
+	const [v3, setV3] = useState(props.query.v3 !== 'false');
+	const [count, setCount] = useState(props.query.count || 20);
+	const [page, setPage] = useState(parseInt(props.query.page) || 1);
+	const [brewCollection, setBrewCollection] = useState(null);
+	const [totalBrews, setTotalBrews] = useState(null);
+	const [searching, setSearching] = useState(false);
+	const [error, setError] = useState(null);
```

Comment:
Most of these states are not used anywhere that I can see, but the same names are used in several functions as parameters. Please rename these or the ones in the function parameters so it is clear when the state is being used and when it is a normal function variable.

Reply:
States are derived from props and used to fill inputs with values from the query. I suppose i could add them directly to the inputs as `defaultValue={props.query.xxx}`

Reply:
My main concern is we have states and function arguments with the same name, so it is difficult to read which one is being used.

Reply:
For example

```
        const updateUrl = (title, author, count, v3, legacy, page) => {
```

This function has several parameters which identical names to the state variables. Either these or the states should probably be renamed so it's not ambiguous.